spec/acceptance/suites/default/89_disable_audit_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'spec_helper_integration'
+
+test_name 'disabling auditing to prep for compliance enforcement'
+
+master      = only_host_with_role(hosts, 'master')
+master_fqdn = fact_on(master, 'fqdn')
+agents      = hosts_with_role(hosts, 'agent')
+
+describe 'disabling auditd to ensure that compliance enforcement works' do
+
+  context 'set up hiera to disable auditing' do
+    let(:prod_env_dir) { '/etc/puppetlabs/code/environments/production' }
+    let(:prod_env_default_yaml) { File.join(prod_env_dir, 'data', 'default.yaml') }
+
+    it 'should disable compliance enforcement and disable auditd' do
+      hiera = YAML.load(on(master, "cat #{prod_env_default_yaml}").stdout)
+
+      # Set profile list to empty string in the existing hiera, because
+      # when no profiles are specified, there is nothing to enforce!
+      hiera['compliance_markup::enforcement'] = []
+
+      # disable auditd
+      hiera['simp_options::auditd'] = true
+      hiera['auditd::enable'] = false
+
+      create_remote_file(master, "#{prod_env_default_yaml}", hiera.to_yaml)
+      on(master, "cat #{prod_env_default_yaml}")
+    end
+  end
+
+  context 'apply auditd-disabling changes' do
+    let(:puppetserver_status_cmd) { puppetserver_status_command(master_fqdn) }
+
+    # can't apply most auditd-related changes until after reboot, but can undo
+    # all the compliance-enforcement settings
+    it 'should undo compliance-enforced settings on agents' do
+      block_on(agents, :run_in_parallel => false) do |agent|
+        # FIXME: By the time we get to the last node, the ssh connection may have been
+        # aggressively terminated by beaker for that node instead of being handled with
+        # reconnect-after-timeout logic.
+        ensure_ssh_connection(agent)
+
+        # We need to do this because the simplib__auditd fact will not be
+        # triggered until after the first puppet run which will cause PAM to
+        # deny access to the system due to the pam_tty_audit bug.
```

Comment:
I don't understand the comment

Reply:
1. `puppet agent -t` => kill auditd
2. You can't SSH in any more because `pam_tty_audit` will kill the session
3. Work around it by running puppet twice in the same run